rfcs/print_test.md

```diff
@@ -0,0 +1,151 @@
+   which do two different things. The former runs the actual print
+   codepath (to PDF) but only does a very partial comparison (page
+   count and (non-rasterirzed) text) of the PDFs. The latter runs a
+   slightly different code-path that results in a sequence of PNGs
+   (one per page) which then get compared. (Note that until 2017 only
+   the latter existed, previously as
+   [`reftest-print`](https://bugzilla.mozilla.org/show_bug.cgi?id=1382327).)
+   (The [bug](https://bugzilla.mozilla.org/show_bug.cgi?id=374050)
+   that introduced `reftest-paged` gives no rational as to why it
+   doesn't use PDF or similar, rather adding a custom code-path; the
+   [bug](https://bugzilla.mozilla.org/show_bug.cgi?id=1299848) that
+   added `print` starts with saying it should compare with the
+   rasterized output but then it gets resolved with very different
+   behaviour.)
+
+* WebKit's history has two halves:
+  * [#15558](https://bugs.webkit.org/show_bug.cgi?id=15558) added
+    support for PDF reftests on 2008-07-09. This was then skipped for
+    the "mac" platform (the one platform it wasn't already skipped on)
+    in [r35135](https://trac.webkit.org/changeset/35135/webkit) two
+    days later due to
+    [#20011](https://bugs.webkit.org/show_bug.cgi?id=20011) (PDFs
+    differ between machines). That bug proposed visually diffing the
+    PDFs instead, but ultimately…
+  * [#37203](https://bugs.webkit.org/show_bug.cgi?id=37203) added
+    support to render in print mode to a PNG (where multiple pages are
+    separated by a horizontal rule). Part of the reason for moving
+    away from PDFs was, "as it would require some amount of efforts to
+    make it portable among [WebKit] ports".
+
+  The latter support survives in both WebKit and Blink, and is triggered by a
+  call to `testRunner.setPrinting()`.
+
+We previously blocked
+[#5381](https://github.com/web-platform-tests/wpt/issues/5381) on
+stopping classifying all the existing print tests as manual, given
+moving all the existing print tests to `-manual` and then having to
+later move them again seems less than ideal, and there was a hope this
+would get solved Real Soon Now® several years ago. We should at least
+agree what we're doing with print tests, even if we don't initially
+have any implementation (and every single runner yields "unsupported
+test type").
+
+## Details
+
+This proposal adds a new test type, called `print-reftest`.
+
+This new test type is similar to the existing `reftest` type; it uses
+`<link rel=match>` and `<link rel=mismatch>`, combining in the same
+way as for normal reftests. It also supports `reftest-wait`. Instead
+of comparing screenshots of the viewport, a print-reftest first
+renders the document to a PDF, and then computes the reftest result by
+rasterising the test and ref PDFs at 96 dpi and comparing the results
```

Comment:
If possible, can we move anything referencing the implementation details under 'Implementation in wptrunner'? In this case, we would instead say that it renders both the test and the ref in print (or paged?) mode (perhaps referencing CSS3 Paged Media), and 'compares the results in a page-by-page...'.

The note about 96 dpi would be moved to implementation details.